Source/Diagnostics/ReducedDiags/FieldReduction.H

```diff
@@ -0,0 +1,184 @@
+/* Copyright 2021 Neil Zaim
+ *
+ * This file is part of WarpX.
+ *
+ * License: BSD-3-Clause-LBNL
+ */
+
+#ifndef WARPX_DIAGNOSTICS_REDUCEDDIAGS_FIELDREDUCTION_H_
+#define WARPX_DIAGNOSTICS_REDUCEDDIAGS_FIELDREDUCTION_H_
+
+#include "ReducedDiags.H"
+#include "WarpX.H"
+#include "Utils/CoarsenIO.H"
+
+/**
+ *  This class contains a function that computes an arbitrary reduction of the fields. The function
+ *  used in the reduction is defined by an input file parser expression and the reduction operation
+ *  can be either Maximum, Minimum, or Sum.
+ */
+class FieldReduction : public ReducedDiags
+{
+public:
+
+    /** constructor
+     *  @param[in] rd_name reduced diags names */
+    FieldReduction(std::string rd_name);
+
+    /** This function is called at everytime step, and if necessary calls the templated function
```

Comment:
```suggestion
    /**
     * This function is called at everytime step, and if necessary calls the templated function
```
I guess that's just a matter of style, but in most cases doxygen comments start with a `/**` followed by a blank line.

Reply:
Just noting here the typo "at everytime step" instead of "at every time step".